src/main/java/org/mule/service/scheduler/internal/DefaultScheduler.java

```diff
@@ -263,29 +294,40 @@ public void setJobClass(Class<? extends QuartzCronJob> jobClass) {
 
   @Override
   public void shutdown() {
-    LOGGER.debug("Shutting down " + this.toString());
-    doShutdown();
-    stopFinally();
+    shutdownLock.writeLock().lock();
+    try {
+      LOGGER.debug("Shutting down " + this.toString());
```

Comment:
I think the toString is not necessary. Verify that the debug is enabled to avoid the logging when not necessary (this is not part of the changes in the PR)

Reply:
Ignore the toString part. It is slightly more efficient to call toString.